services/bitbucket/bitbucket-pull-request.service.js

```diff
@@ -24,15 +25,57 @@ function pullRequestClassGenerator(raw) {
       })
     }
 
+    async fetchServer({ proto, hostAndPath, user, repo }) {
+      const url = `${proto}://${hostAndPath}/rest/api/1.0/projects/${user}/repos/${repo}/pull-requests`
+      const options = {
+        qs: {
+          state: 'OPEN',
+          limit: 100,
+          withProperties: false,
+          withAttributes: false,
+        },
+      }
+
+      if (
+        serverSecrets &&
+        serverSecrets.bitbucket_server_username &&
+        serverSecrets.bitbucket_server_password
+      ) {
+        options.auth = {
+          user: serverSecrets.bitbucket_server_username,
+          pass: serverSecrets.bitbucket_server_password,
+        }
+      }
+
+      return this._requestJson({
+        url,
+        schema: bitbucketPullRequestsSchema,
+        options,
+        errorMessages: {
+          401: 'Authentication Required',
+          403: 'Private Repo',
+          404: 'Repo Not Found',
+        },
+      })
+    }
+
+    async fetch({ proto, hostAndPath, user, repo }) {
+      if (proto !== undefined && hostAndPath !== undefined) {
```

Comment:
This doesn't necessarily need to be changed, but with the regex defined in the route format we don't need to check both of these (if one has a value then the other will always be guaranteed to have a value as well)

Reply:
Makes sense